src/modules/itslearning/event-handlers/itslearning-persons.event-handler.ts

```diff
@@ -36,60 +42,86 @@ export class ItsLearningPersonsEventHandler {
+        await this.personUpdateMutex.runExclusive(async () => {
+            this.logger.info(`Received PersonRenamedEvent, ${event.personId}`);
 
-        if (!this.ENABLED) {
-            return this.logger.info('Not enabled, ignoring event.');
-        }
+            if (!this.ENABLED) {
+                return this.logger.info('Not enabled, ignoring event.');
+            }
 
-        if (!event.referrer) {
-            return this.logger.error(`Person with ID ${event.personId} has no username!`);
-        }
+            if (!event.referrer) {
+                return this.logger.error(`Person with ID ${event.personId} has no username!`);
+            }
+
+            const readPersonResult: Option<PersonResponse> = await this.itslearningPersonRepo.readPerson(
+                event.personId,
+            );
 
-        const readPersonResult: Option<PersonResponse> = await this.itslearningPersonRepo.readPerson(event.personId);
+            if (!readPersonResult) {
+                return this.logger.info(`Person with ID ${event.personId} is not in itslearning, ignoring.`);
+            }
 
-        if (!readPersonResult) {
-            return this.logger.info(`Person with ID ${event.personId} is not in itslearning, ignoring.`);
-        }
+            const updatePersonError: Option<DomainError> = await this.itslearningPersonRepo.createOrUpdatePerson({
+                id: event.personId,
+                firstName: event.vorname,
+                lastName: event.familienname,
+                username: event.referrer,
+                institutionRoleType: readPersonResult.institutionRole,
+            });
 
-        const updatePersonError: Option<DomainError> = await this.itslearningPersonRepo.createOrUpdatePerson({
-            id: event.personId,
-            firstName: event.vorname,
-            lastName: event.familienname,
-            username: event.referrer,
-            institutionRoleType: readPersonResult.institutionRole,
+            if (updatePersonError) {
+                return this.logger.error(`Person with ID ${event.personId} could not be updated in itsLearning!`);
+            }
+
+            this.logger.info(`Person with ID ${event.personId} updated in itsLearning!`);
         });
+    }
 
-        if (updatePersonError) {
-            return this.logger.error(`Person with ID ${event.personId} could not be updated in itsLearning!`);
-        }
+    @EventHandler(OxUserChangedEvent)
+    public async oxUserChangedEventHandler(event: OxUserChangedEvent): Promise<void> {
+        await this.personUpdateMutex.runExclusive(async () => {
+            this.logger.info(`Received OxUserChangedEvent, ${event.personId}`);
+
+            await this.updateEmail(event.personId, event.primaryEmail);
+        });
+    }
 
-        this.logger.info(`Person with ID ${event.personId} updated in itsLearning!`);
+    @EventHandler(OxMetadataInKeycloakChangedEvent)
```

Comment:
Is it not the same as @EventHandler(OxUserChangedEvent)? When a user changes in ox, the user ifno will be updated in KC? Or does ox have another kc instance :)?

Reply:
When I implemented this I asked @DPDS93CT if I need both, but I'll check if it is actually needed

Reply:
`OxUserChangedEvent` seems to be the only relevant event, the other one is thrown inside of an `OxUserChangedEvent`-handler. I've removed the second event-handler so we just have the `OxUserChangedEvent`-handler